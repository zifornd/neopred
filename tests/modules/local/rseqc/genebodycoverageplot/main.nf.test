nextflow_process {

    name "Test Process RSEQC_GENEBODYCOVERAGEPLOT"
    script "modules/local/rseqc/genebodycoverageplot/main.nf"
    process "RSEQC_GENEBODYCOVERAGEPLOT"

    test("Running gene body coverage plot with -stub options") {
          setup{
	     run("RSEQC_GENEBODYCOVERAGE") {
		script "../../../../../modules/local/rseqc/genebodycoverage/main.nf"
		process {
			"""
			input[0] = Channel.of([ [ id:'test' ], file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true), file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)])
	               input[1] = file('${projectDir}/assets/housekeeping_refseqGenes.bed')
     	 		"""
		        }
	      }
	}
        when {
	options "-stub"
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = RSEQC_GENEBODYCOVERAGE.out.rscript
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
