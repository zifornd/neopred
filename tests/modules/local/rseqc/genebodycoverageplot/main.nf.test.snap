{
    "Running gene body coverage plot with -stub options": {
        "content": [
            {
                "0": [
                    "geneBodyCoverage.r:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "geneBodyCoverage.curves.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,b6aed5d7b360967b1abb1ff6920a6815"
                ],
                "g_rscript": [
                    "geneBodyCoverage.r:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "png_curves": [
                    "geneBodyCoverage.curves.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,b6aed5d7b360967b1abb1ff6920a6815"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-07T07:04:44.370476"
    }
}